### DISCUSSION TITLE
how to add if/else Api rules for updating a 'posts' collection

### DISCUSSION THREAD

**Discussion by phyoehtataung** [pocketbase/pocketbase#1661] [Q&A] [answered]
Hi, so my 'posts' collection has 'title', 'body', 'likes: { }' , etc...

I want to set rules such that updating a post likes can be done by any auth user but updating other fields must be done by post author:

if @request.data.likes then @request.auth.id != "" else @request.auth.id = author

how to achieve that? Thanks in advance 😄

**Comment by ganigeorgiev** [comment, marked as answer]
You can use the `&&` (and) and `||` (or) operators, something like:

```js
(@request.data.likes:isset = true && @request.auth.id != "") || @request.auth.id = author
```

For more complex access controls, you can also make use of the `OnRecordBefore*` request hooks - https://pocketbase.io/docs/event-hooks/#record-api-hooks (_in a future version there will be also an option to bind to the app hooks with js_).

**Reply by ganigeorgiev** [reply]
Actually, on second read, please note that the above rule example was "direct translation" of the `if/then/else` from the question. But if you want to restrict fields update, you'll have to manually check all other fields (either with `:isset` or with a fixed value):

```js
author = @request.auth.id ||
(
   @request.auth.id != "" &&
   @request.data.otherField1:isset = false &&
   @request.data.otherField2:isset = false
   ...
)
```

**Reply by phyoehtataung** [reply]
thank you for reply! 😄
